src/App.tsx

```diff
@@ -1,26 +1,234 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  USER_ID,
+  addTodos,
+  deleteTodos,
+  getTodos,
+  updateTodos,
+} from './api/todos';
+import { Header } from './Components/Header/Header';
+import { Todo } from './types/Todo';
+import { Footer } from './Components/Footer/Footer';
+import { TodoList } from './Components/TodoList/TodoList';
+import { Errors } from './Components/Errors/Errors';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState(Status.All);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState<number[]>([]);
+  const [inputTitle, setInputTitle] = useState('');
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputTitle(event.target.value);
+  };
+
+  const itemsLeft = todos.filter(todo => !todo.completed).length;
+
+  const clearAllButtonActive = todos.length - itemsLeft;
+
+  function onAdd({ userId, completed }: Todo) {
+    setInputDisabled(true);
+    const trimmedTitle = inputTitle.trim();
+
+    if (!trimmedTitle) {
+      setErrorMessage('Title should not be empty');
+      setInputDisabled(false);
+      setTimeout(() => setErrorMessage(''), 3000);
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: trimmedTitle,
+      completed: false,
+    });
+
+    setIsLoading(current => [...current, 0]);
+
+    addTodos({ userId, title: trimmedTitle, completed })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setInputTitle('');
+      })
+      .catch(error => {
+        setErrorMessage('Unable to add a todo');
+        setInputDisabled(false);
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+        throw error;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)